packages/flutter_tools/lib/src/asset.dart

```diff
@@ -425,7 +434,23 @@ class ManifestAssetBundle implements AssetBundle {
 
     _setIfChanged(_kAssetManifestJson, assetManifest);
     _setIfChanged(kFontManifestJson, fontManifest);
-    _setIfChanged(_kNoticeFile, licenses);
+    if (targetPlatform == TargetPlatform.web_javascript) {
```

Comment:
I would split up this code into a method with good documentation on what it is doing. Its a bit hard to follow the conditional as is